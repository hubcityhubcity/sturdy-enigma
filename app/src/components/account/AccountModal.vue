<template>
  <transition name="component-fade" mode="out-in">
    <div
      v-focus-last
      class="session-frame"
      tabindex="0"
      @keyup.esc="close()"
      @click.self="close()"
    >
      <div class="session-outer-container">
        <div class="session">
          <div class="session-header">
            <a @click="close()">
              <i class="material-icons notranslate circle back">arrow_back</i>
            </a>
            <div v-if="!isExtension" class="session-close">
              <a @click="close()">
                <i class="material-icons notranslate circle back">close</i>
              </a>
            </div>
          </div>
          <slot>
            <h1 class="session-title">Feature Not Available</h1>
            <p>(But coming soon)</p>
            <!-- common form for account modals -->
            <!-- TODO: hide for now -->
            <TmFormGroup
              class="account-modal-form-group"
              field-id="password"
              style="display: none;"
            >
              <TmField
                id="password"
                v-model="password"
                type="password"
                placeholder="Password"
              />
              <div class="account-modal-buttons">
                <TmBtn
                  value="Dismiss"
                  type="secondary"
                  @click.native="close"
                  @click.enter.native="close"
                />
                <TmBtn
                  value="Reveal"
                  type="primary"
                  @click.native="revealSeedPhrase"
                  @click.enter.native="revealSeedPhrase"
                />
              </div>
            </TmFormGroup>
          </slot>
        </div>
      </div>
    </div>
  </transition>
</template>
<script>
import TmFormGroup from "common/TmFormGroup"
import TmField from "common/TmField"
import TmBtn from "common/TmBtn"
import config from "src/../config"
import { mapGetters } from "vuex"
export default {
  name: `account-modal`,
  components: {
    TmFormGroup,
    TmField,
    TmBtn,
  },
  data: () => {
    return {
      password: "",
      isExtension: config.isExtension,
    }
  },
  computed: {
    ...mapGetters([`networkSlug`]),
  },
  methods: {
    close() {
      this.$router.push({
        name: "portfolio",
        params: {
          networkId: this.networkSlug,
        },
      })
    },
  },
}
</script>
<style>
@import "../../styles/session.css";

.component-fade-enter-active,
.component-fade-leave-active {
  transition: opacity 0.2s ease;
}

.component-fade-enter,
.component-fade-leave-to .component-fade-leave-active {
  opacity: 0;
}
</style>

<style scoped>
.account-modal-buttons {
  display: flex;
}

.session-main p {
  text-align: center;
}
</style>
