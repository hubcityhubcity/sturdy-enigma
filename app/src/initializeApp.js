/* istanbul ignore file: really just integrations */

import { checkForNewLunieVersions } from "scripts/check-for-new-lunie-versions"
import {
  enableGoogleAnalytics,
  setGoogleAnalyticsPage,
} from "scripts/google-analytics"
import config from "src/../config"
import Router, { routeGuard } from "./router"
import Store from "./vuex/store"

if (navigator && navigator.serviceWorker) {
  // remove any existing service worker
  navigator.serviceWorker.getRegistrations().then(function (registrations) {
    for (let registration of registrations) {
      registration.unregister()
    }
  })
}

export const bootError = (error) => {
  document.querySelector("#app").innerHTML = `
  <div class="boot-error">
    <h2>Uh oh. Lunie failed to start. Our team has been notified of the issue. Please try refreshing this page in a few minutes. If the error persists please contact us at http://help.lunie.io/</h2>

    <p class="error">${error}</p>
  </div>
    `
}

export default async function init(urlParams, env = process.env) {
  // add error handlers in production
  if (env.NODE_ENV === `production`) {
    // check every minute if new Lunie versions have been deployed
    checkForNewLunieVersions()
    enableGoogleAnalytics(config.google_analytics_uid)
  }

  const store = Store()

  const router = Router(store)
  setGoogleAnalyticsPage(router.currentRoute.path)

  store.dispatch(`loadLocalPreferences`)
  // store.dispatch(`checkForPersistedAddresses`)

  return { store, router }
}
