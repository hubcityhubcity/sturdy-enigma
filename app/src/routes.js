/**
 * In this module we took care of the definition of our routes, with parameters, children and component related to them
 * @module routes
 */

/**
 * Routes are all defined here
 */
/* istanbul ignore file */
export default (store) => {
  return [
    {
      path: `/reveal/:address`,
      name: `reveal`,
      component: require("account/RevealSeedModal").default,
    },
    {
      path: `/delete/:address/:addressNetworkId`,
      name: `delete`,
      component: require("account/ForgetAccountModal").default,
    },
    {
      path: `/`,
      name: `manage-accounts-modal`,
      component: () => import(`./components/account/ManageAccountsModal`),
      meta: {
        networkSpecificRoute: true,
      },
    },
  ]
}

// handle direct sign in from the extension via deeplink
export async function extensionSignIn({ to, next }, store) {
  let network = store.getters.networks.find(
    ({ id }) => id === to.params.network
  )

  await store.dispatch(`signIn`, {
    sessionType: `extension`,
    address: to.params.address,
    networkId: to.params.network,
  })
  next(`/${network.slug}/portfolio`)
}
