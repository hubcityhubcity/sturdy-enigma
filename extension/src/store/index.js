import Vue from 'vue'
import Vuex from 'vuex'
import Vuelidate from 'vuelidate'

import * as getters from './getters'
import * as mutations from './mutations'
import actions from './actions'

Vue.use(Vuex)
Vue.use(Vuelidate)

export default (opts = {}) => {
  return new Vuex.Store({
    state: {
      accounts: [],
      signRequest: null,
      session: {
        insecureMode: true
      },
      signup: {
        signUpName: ``,
        signUpPassword: ``,
        signUpPasswordConfirm: ``,
        signUpWarning: false,
        signUpSeed: ``
      },
      recover: {
        seed: ``,
        name: ``,
        password: ``,
        passwordConfirm: ``
      },
      network: '',
      networkSlug: `cosmos-hub`,
      networks: []
    },
    getters,
    actions: actions(opts),
    mutations
  })
}
