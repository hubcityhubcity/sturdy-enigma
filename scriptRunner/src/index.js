var express = require('express')
var bodyParser = require('body-parser')
var timeout = require('connect-timeout');
var morgan = require('morgan')
var app = express()
require('dotenv').config()
const config = require('../config')

// Constants
const PORT = process.env.PORT || 9000;
const HOST = process.env.NODE_ENV = "docker" ? '0.0.0.0' : 'localhost';

const polkadotrewards = require("./polkadotrewards")

app.use(bodyParser.json())
app.use(timeout(120000))
app.use(morgan('combined'))
app.use(function (req, res, next) {
    const authenticationToken = req.header("Authorization")
    if (authenticationToken !== config.authenticationToken) {
        res
            .status(403)
            .send()
        return
    }
    next()
})


app.post('/polkadotrewards', polkadotrewards)

app.listen(PORT, HOST, () => console.log(`Script server running on http://localhost:${PORT}`))