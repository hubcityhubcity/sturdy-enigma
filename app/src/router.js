import router from "vue-router"
import routes from "./routes"
import Vue from "vue"

/* istanbul ignore next */
Vue.use(router)

/* istanbul ignore next */
const Router = (store) =>
  new router({
    mode: process.env.VUE_APP_E2E ? undefined : "history",
    routes: routes(store),
  })

export default Router
