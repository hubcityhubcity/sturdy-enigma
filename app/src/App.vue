<template>
  <div id="app" :class="network">
    <NetworkSelector />
    <AppHeader />
    <div id="app-content">
      <div id="bar-container">
        <CookieBar v-if="!isMobileApp" />
        <MaintenanceBar />
        <DisconnectedBar />
      </div>
      <div class="user-controls">
        <TmBtn
          id="niteModeBtn"
          :value="isNiteMode ? `☀️ DayMode` : `🌜 NiteMode`"
          type="secondary"
          @click.native="niteModeToggle()"
        />
        <UserMenu v-if="!isMobileApp && session.experimentalMode" />
      </div>
      <router-view name="session" />
      <router-view />
    </div>
    <MobileMenu />
  </div>
</template>

<script>
import { mapState, mapGetters } from "vuex"
import AppHeader from "common/AppHeader"
import CookieBar from "common/CookieBar"
import DisconnectedBar from "common/DisconnectedBar"
import UserMenu from "account/UserMenu"
import MaintenanceBar from "common/MaintenanceBar"
import MobileMenu from "common/MobileMenu"
import NetworkSelector from "common/NetworkSelector"
import TmBtn from "src/components/common/TmBtn"
import store from "./vuex/store"

export default {
  name: `app`,
  components: {
    AppHeader,
    CookieBar,
    DisconnectedBar,
    UserMenu,
    MaintenanceBar,
    MobileMenu,
    NetworkSelector,
    TmBtn,
  },
  data: () => ({
    isNiteMode: false,
  }),
  computed: {
    ...mapState([`notifications`, `session`]),
    ...mapGetters([`network`]),
    isMobileApp() {
      return this.session.mobile
    },
  },
  methods: {
    niteModeToggle() {
      if (this.isNiteMode) {
        this.$el.classList = [this.network]
        this.isNiteMode = false
      } else {
        this.$el.classList = [`lunie-dark`]
        this.isNiteMode = true
      }
    },
  },
  store,
}
</script>

<style>
@import "./styles/app.css";

.user-controls {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

#niteModeBtn {
  width: 4rem;
  margin-left: 1rem;
}
</style>
