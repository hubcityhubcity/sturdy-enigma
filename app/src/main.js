/* istanbul ignore file */

import Vue from "vue"
import VTooltip from "v-tooltip"
import Vuelidate from "vuelidate"
import VueClipboard from "vue-clipboard2"
import { focusParentLast } from "src/directives"
import App from "./App.vue"
import init, { bootError } from "./initializeApp"
import { getURLParams } from "scripts/url"
import { Plugins } from "@capacitor/core"
import config from "src/../config"
import * as Sentry from "@sentry/browser"
import * as Integrations from "@sentry/integrations"
import "material-design-icons-iconfont/dist/material-design-icons.css"
import AsyncComputed from "vue-async-computed"

if (config.sentryDSN) {
  Sentry.init({
    dsn: config.sentryDSN,
    integrations: [new Integrations.Vue({ Vue, attachProps: true })],
    release: process.env.LUNIE_VERSION,
  })
}

Vue.config.productionTip = false

Vue.use(VTooltip)

Vue.use(Vuelidate)
Vue.use(VueClipboard)
Vue.use(AsyncComputed)

Vue.directive(`focus-last`, focusParentLast)

const urlParams = getURLParams(window)
init(urlParams)
  .then(({ store, router }) => {
    const { SplashScreen, StatusBar } = Plugins

    new Vue({
      router,
      ...App,
      store,
      mounted() {
        if (config.mobileApp) {
          SplashScreen.hide()
          StatusBar.show()
        }
      },
    }).$mount("#app")
  })
  .catch((error) => {
    Sentry.captureException(error)
    bootError(error)
  })
