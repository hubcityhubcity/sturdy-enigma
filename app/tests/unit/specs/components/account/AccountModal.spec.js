import { shallowMount, createLocalVue } from "@vue/test-utils"
import AccountModal from "src/components/account/AccountModal"
import { focusParentLast } from "src/directives"

const localVue = createLocalVue()
localVue.directive("focus-last", focusParentLast)

describe(`AccountModal`, () => {
  let wrapper, $store

  beforeEach(() => {
    $store = {
      getters: {
        networkSlug: `cosmos`,
      },
    }
    wrapper = shallowMount(AccountModal, {
      localVue,
      mocks: {
        $store,
        $router: {
          push: jest.fn(),
        },
      },
    })
  })

  it(`should show the AccountModal page`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
