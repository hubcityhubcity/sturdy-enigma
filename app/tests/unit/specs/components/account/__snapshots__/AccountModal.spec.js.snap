// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountModal should show the AccountModal page 1`] = `
<div
  class="session-frame"
  mode="out-in"
  name="component-fade"
  tabindex="0"
>
  <div
    class="session-outer-container"
  >
    <div
      class="session"
    >
      <div
        class="session-header"
      >
        <a>
          <i
            class="material-icons notranslate circle back"
          >
            arrow_back
          </i>
        </a>
         
        <div
          class="session-close"
        >
          <a>
            <i
              class="material-icons notranslate circle back"
            >
              close
            </i>
          </a>
        </div>
      </div>
       
      <h1
        class="session-title"
      >
        Feature Not Available
      </h1>
       
      <p>
        (But coming soon)
      </p>
       
      <tmformgroup-stub
        class="account-modal-form-group"
        fieldid="password"
        style="display: none;"
      >
        <tmfield-stub
          id="password"
          placeholder="Password"
          type="password"
          value=""
        />
         
        <div
          class="account-modal-buttons"
        >
          <tmbtn-stub
            customclass=""
            type="secondary"
            value="Dismiss"
          />
           
          <tmbtn-stub
            customclass=""
            type="primary"
            value="Reveal"
          />
        </div>
      </tmformgroup-stub>
    </div>
  </div>
</div>
`;
